lib/dogapi/common.rb

```diff
@@ -164,10 +164,9 @@ def Dogapi.find_datadog_host
 
   def Dogapi.find_localhost
     begin
-      # prefer hostname -f over Socket.gethostname
-      @@hostname ||= %x[hostname -f].strip
+      @@hostname ||= Addrinfo.getaddrinfo(Socket.gethostname, nil, nil, nil, nil, Socket::AI_CANONNAME).first.canonname
     rescue
-      raise 'Cannot determine local hostname via hostname -f'
+      raise 'Cannot determine local hostname via Socket.gethostname'
```

Comment:
While we're here, I think it may be helpful to also provide the message of the original error that caused this to occur. What do you think?